### DISCUSSION TITLE
Property Schema - Applies to Option  Query

### DISCUSSION THREAD

**Discussion by rsrinivasanhome** [simonec73/threatsmanager#23] [Q&A] [answered]
I have now migrated my template in threat manager studio and performed the needed corrections.  I have a doubt on the functionality
of "Applies To Property list" [feature](https://threatsmanager.com/training/expert/metadata/property-schemas/) of threats manager studio.
![image](https://user-images.githubusercontent.com/31912418/148202092-f95296aa-90fb-4f69-931f-16fa2f2c51ea.png)

In the screen shot above  I am adding a new property and press apply schema button. 
Is it meant to apply the schema on all external Interactors or only the particular property schema ?  What happens when I make "Applies to " empty ?

**Comment by simonec73** [comment, marked as answer]
The **Applies To** field determines what objects will receive the Property Schema. If you leave it empty, no object will receive it, ever.
**Apply Automatically** flag determines if new objects receive it. Button **Apply Schema** in the Ribbon allows assigning the Property Schema to existing objects.

So, even if you check the flag **Apply Automatically** in the example, existing objects will not receive the new property "Network security enabled". To ensure they get it, you have to click the button **Apply Schema**. 

The latest versions have also introduced another possibility. Let's imagine that you do not want to assign the Property Schema to all External Interactors, but just to a few you specifically decide. You can do that by selecting the External Interactor you want to receive the Property Schema, then open the context menu (right mouse button on the object), and select **Apply Property Schema**. This will open a dialog that allows you to select the Property Schema to apply.
![image](https://user-images.githubusercontent.com/23357188/148220939-6bbb9915-d85c-4432-8ac3-d78630e486ef.png)

Please note that there is also a **Remove Property Schema** action you can use to selectively remove unnecessary Property Schemas.

I hope it helps. I understand that it may seem complicated at first, but I hope you'll appreciate the power you get at your fingertips. If it seems too much, please consider that all those capabilities are available from Export Execution Mode onward: non-experts Threat Modelers would work with Simplified mode, which does hide this complexity.
